algebra/src/fft/evaluations.rs

```diff
@@ -59,7 +57,7 @@ impl<'a, 'b, F: PrimeField> Mul<&'a Evaluations<F>> for &'b Evaluations<F> {
 impl<'a, F: PrimeField> MulAssign<&'a Evaluations<F>> for Evaluations<F> {
     #[inline]
     fn mul_assign(&mut self, other: &'a Evaluations<F>) {
-        assert_eq!(self.domain, other.domain, "domains are unequal");
+        assert_eq!(self.domain.as_ref(), other.domain.as_ref(), "domains are unequal");
```

Comment:
Here, and as well below: let's parallelize these vector operations (arkworks did that already).